Java/core/src/main/java/com/amazon/randomcutforest/tree/IBoundingBox.java

```diff
@@ -29,13 +29,25 @@
 
     boolean contains(P point);
 
-    IBox<P> getMergedBox(P point);
+    // duplicates
+    IBoundingBox<P> copy();
 
-    IBox<P> getMergedBox(IBox<P> otherBox);
+    // below keeps the older box unchanged
+    IBoundingBox<P> getMergedBox(P point);
 
-    BoundingBox convertBoxToDouble();
+    // merges in place
+    IBoundingBox<P> addPoint(P point);
 
-    BoundingBoxFloat convertBoxToFloat();
+    // merges and keeops the older box unchaged
+    IBoundingBox<P> getMergedBox(IBoundingBox<P> otherBox);
+
+    // merges in place
+    IBoundingBox<P> addBox(IBoundingBox<P> otherBox);
+
+    BoundingBox copyBoxToDouble();
```

Comment:
Would everything work if the return type of this method was `IBoundingBox<double[]>`?

Reply:
At the current moment the Visitors need BoundingBox -- we should change that in a subsequent PR.

Reply:
https://github.com/aws/random-cut-forest-by-aws/issues/99

Reply:
Can you add a TODO comment to remove this method?